### PR TITLE
fix(llm): fix carousel pagination for content cards

### PR DESCRIPTION
**Pull request by LucasWerey**
This PR regroups those two tickets : [LIVE-11392](https://ledgerhq.atlassian.net/browse/LIVE-11392) & [LIVE-11414](https://ledgerhq.atlassian.net/browse/LIVE-11414)

You can check the following PR what has been merged in here : https://github.com/LedgerHQ/ledger-live/pull/6533
### ✅ Checklist

- [x] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [x] **Impact of the changes:** 
  - LLM
  - Braze

### 📝 Description

Fix pagination bug for carousel generated from Braze.
It will jumps to the last indicator with small actions cards

| Before        | After         |
| ------------- | ------------- |
|  https://github.com/LedgerHQ/ledger-live/assets/73439207/110055f7-8876-48ef-8098-0a4405fb0727|  https://github.com/LedgerHQ/ledger-live/assets/73439207/8226cc39-9ead-4309-9931-a0c15a2bb81a
|image|image |

### ❓ Context

- **JIRA or GitHub link**: [LIVE-11414](https://ledgerhq.atlassian.net/browse/LIVE-11414)

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-11414]: https://ledgerhq.atlassian.net/browse/LIVE-11414?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[LIVE-11392]: https://ledgerhq.atlassian.net/browse/LIVE-11392?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-mobile/src/dynamicContent/types.ts [hunk 1]
```diff
@@ -51,7 +51,7 @@ type CategoryContentCard = ContentCardCommonProperties & {
   cta?: string;
   link?: string;
   isDismissable?: boolean;
-  pagination?: boolean;
+  hasPaggination?: boolean;
```
Comment (KVNLS) [inline apps/ledger-live-mobile/src/dynamicContent/types.ts:54]: Typo: `hasPagination`

File: apps/ledger-live-mobile/src/dynamicContent/utils.tsx [hunk 2]
```diff
@@ -79,6 +79,7 @@ export const mapAsCategoryContentCard = (card: BrazeContentCard): CategoryConten
   link: card.extras.link,
   cta: card.extras.cta,
   isDismissable: Boolean(card.extras.isDismissable && card.extras.isDismissable === "true"),
+  hasPaggination: Boolean(card.extras.hasPaggination && card.extras.hasPaggination === "true"),
```
Comment (KVNLS) [inline apps/ledger-live-mobile/src/dynamicContent/utils.tsx:82]: You can use optional chaining https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining
```suggestion
  hasPaggination: Boolean(card.extras?.hasPaggination === "true"),
```
If `extras` is optional you can do the same:
`card?.extras?.hasPaggination === "true"`

**Review by KVNLS — CHANGES_REQUESTED** [review]

File: apps/ledger-live-mobile/src/dynamicContent/utils.tsx (hunk 2 above)
Comment (LucasWerey) [reply apps/ledger-live-mobile/src/dynamicContent/utils.tsx]: Done ✅

File: apps/ledger-live-mobile/src/dynamicContent/types.ts (hunk 1 above)
Comment (LucasWerey) [reply apps/ledger-live-mobile/src/dynamicContent/types.ts]: Done ✅

**Review by gre — CHANGES_REQUESTED** [review]
please fill up the PR checklist 🙏 
Capture d’écran 2024-03-22 à 18 14 14

**Review by sshmaxime — APPROVED** [review]

**Pull request merged**